src/swap/components/SwapSettings.tsx

```diff
@@ -0,0 +1,37 @@
+import { isValidElement, useMemo } from 'react';
+import { SwapSettingsSvg } from '../../internal/svg/swapSettings';
+import { cn } from '../../styles/theme';
+import type { SwapSettingsReact } from '../types';
+
+export function SwapSettings({
+  className,
+  icon,
+  title = 'Auto',
+}: SwapSettingsReact) {
+  const iconSvg = useMemo(() => {
+    if (icon === undefined) {
+      return SwapSettingsSvg;
+    }
+    if (isValidElement(icon)) {
+      return icon;
+    }
+  }, [icon]);
+
+  return (
+    <div
```

Comment:
So we have three `div` here.
```
<div className="flex justify-end">
  <div  className={cn('flex w-full items-center justify-between', className)}>
     <div className="flex items-center space-x-1">
 ```

do we really need all of them? also should holistically have a better semantic?

Reply:
Refactored this to have one div